pennylane/workflow/execution.py

```diff
@@ -801,9 +819,16 @@ def device_gradient_fn(inner_tapes, **gradient_kwargs):
         jpc = TransformJacobianProducts(
             execute_fn, gradient_fn, gradient_kwargs, cache_full_jacobian
         )
-        for _ in range(1, max_diff):
-            ml_boundary_execute = _get_ml_boundary_execute(interface, _grad_on_execution)
-            execute_fn = partial(ml_boundary_execute, execute_fn=execute_fn, jpc=jpc, device=device)
+        for i in range(1, max_diff):
+            ml_boundary_execute = _get_ml_boundary_execute(
+                interface, _grad_on_execution, differentiable=i > 1
+            )
+            execute_fn = partial(

```

Comment:
Minor: Can you get `i > 1` out and assign it to a variable?

Reply:
assigned to `differentiable`.